cumulusci/tasks/salesforce/sourcetracking.py

```diff
@@ -0,0 +1,109 @@
+from cumulusci.tasks.salesforce import BaseRetrieveMetadata
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.tasks.metadata.package import PackageXmlGenerator
+from cumulusci.tasks.metadata.package import __location__
+
+
+class ListChanges(BaseSalesforceApiTask):
+    def _run_task(self):
+        changes = self.tooling.query(
+            "SELECT MemberName, MemberType FROM SourceMember WHERE IsNameObsolete=false"
+        )
+        if changes["totalSize"]:
+            self.logger.info(
+                "Found {} changed components in the scratch org:".format(
+                    changes["totalSize"]
+                )
+            )
+            for change in changes["records"]:
+                self.logger.info(
+                    "  {}: {}".format(change["MemberType"], change["MemberName"])
+                )
+        else:
+            self.logger.info("Found no changes.")
+
+
+class RetrieveChanges(BaseRetrieveMetadata, BaseSalesforceApiTask):
+    api_class = ApiRetrieveUnpackaged
+
+    task_options = {
+        "path": {
+            "description": "The path to write the retrieved metadata",
+            "required": True,
+        },
+        "include": {
+            "description": "Include changed components matching this string.",
+            "required": True,
+        },
+        "api_version": {
+            "description": (
+                "Override the default api version for the retrieve."
+                + " Defaults to project__package__api_version"
+            )
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super(RetrieveChanges, self)._init_options(kwargs)
+
+        if "api_version" not in self.options:
+            self.options[
+                "api_version"
+            ] = self.project_config.project__package__api_version
+
+        self.options["include"] = self.options["include"].split(" ")
+
+    def _get_api(self):
+        self.logger.info("Querying Salesforce for changed source members")
+        changes = self.tooling.query(
+            "SELECT MemberName, MemberType FROM SourceMember WHERE IsNameObsolete=false"
+        )
+
+        type_members = defaultdict(list)
+        for change in changes["records"]:
+            mdtype = change["MemberType"]
+            name = change["MemberName"]
+            if any(s in mdtype or s in name for s in self.options["include"]):
+                self.logger.info("Including {} ({})".format(name, mdtype))
+                type_members[mdtype].append(name)
+
+        types = []
+        for name, members in type_members.items():
+            types.append(MetadataType(name, members))
+        package_xml = PackageXmlGenerator(self.options["api_version"], types=types)()
+
+        return self.api_class(self, package_xml, self.options.get("api_version"))
+
+    def _run_task(self):
+        super(RetrieveChanges, self)._run_task()
+
+        # update package.xml
```

Comment:
hmm, feels like we should just use normal update_package_xml task as the next step in a flow, instead of duplicating it?

Reply:
Hmm. Is there a use case where you would want to retrieve changes but not update package.xml? I have been thinking no, and thus thinking that we should avoid having a task which doesn't do what you want unless you realize it's meant to be used in a particular flow.

Reply:
👍